src/Nodes1DProvisioner.cpp

```diff
@@ -71,6 +79,86 @@ void Nodes1DProvisioner::buildNodes() {
     }
 
     buildConnectivityMatrices();
+    buildFaceMask();
+}
+
+/**
+ * Build volume to surface maps.
+ */
+void Nodes1DProvisioner::buildMaps() {
+    firstIndex ii;
+    secondIndex jj;
+
+    index_matrix_type nodeIds(NumLocalPoints, NumElements);
+
+    // Set up reference to the objects we need to interact with.
+    SparseMatrixConverter & matConverter = *MatrixConverter;
+    
+    index_vector_type & Fmsk = *Fmask;
+    index_vector_type & vmM = *vmapM;
+    index_vector_type & vmP = *vmapP;
+
+    index_matrix_type & E2E = *EToE;
+    index_matrix_type & E2F = *EToF;
+    matrix_type & xmat = *xGrid;
+
+    matrix_type xmatTrans(NumElements, NumLocalPoints);
+    xmatTrans = xmat(jj,ii);
+
+    double * x = new double[NumElements*NumLocalPoints];
+    matConverter.fullToPodArray(xmatTrans, x);
```

Comment:
Does fullToPodArray take ownership of the memory allocated to x? If not, then you forgot to delete this memory.

Reply:
Good catch!